sdk/opa_test.go

```diff
@@ -215,6 +216,62 @@ func TestDecision(t *testing.T) {
 	}
 }
 
+func TestDecisionWithStrictBuiltinErrors(t *testing.T) {
+
+	ctx := context.Background()
+
+	server := sdktest.MustNewServer(
+		sdktest.MockBundle("/bundles/bundle.tar.gz", map[string]string{
+			"main.rego": `
+				package example
+
+erroring_function(number) = output {
+	output := number / 0
+}
+
+allow {
+	erroring_function(1)
+}`,
+		}),
+	)
+
+	defer server.Stop()
+
+	config := fmt.Sprintf(`{
+		"services": {
+			"test": {
+				"url": %q
+			}
+		},
+		"bundles": {
+			"test": {
+				"resource": "/bundles/bundle.tar.gz"
+			}
+		}
+	}`, server.URL())
+
+	opa, err := sdk.New(ctx, sdk.Options{
+		Config: strings.NewReader(config),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer opa.Stop(ctx)
+
+	_, err = opa.Decision(ctx, sdk.DecisionOptions{
+		StrictBuiltinErrors: true,
+		Path:                "/example/allow",
+	})
+	if err == nil {
+		t.Fatal("expected error but got nil")
+	}
+
+	if actual, ok := err.(*topdown.Error); !ok || actual.Code != "eval_builtin_error" {
+		t.Fatalf("expected eval_builtin_error but got %v", actual)
```

Comment:
Since one point of enabling this would be to see what built-in failed — with their message — I'd love to see that asserted here.

Reply:
I.e. not just the type of error but the actual built-in error message.

Reply:
Thanks, that makes sense. I've added an assertion for the message content in a2be7c46bef8ec78d2a2d4a14a60a414581d17b4.

Reply:
Nice, thanks!